tests/test_dataset.py

```diff
@@ -0,0 +1,28 @@
+"""High level tests for Rasterio's ``GDALDataset`` abstractions."""
+
+
+import os
+
+import pytest
+
+import rasterio
+from rasterio.errors import RasterioIOError
+
+
+def test_files(path_rgb_byte_tif):
+    aux = path_rgb_byte_tif + '.aux.xml'
+    with open(aux, 'w'):
+        pass
+    with rasterio.open(path_rgb_byte_tif) as src:
+        assert src.files == (path_rgb_byte_tif, aux)
+    os.unlink(aux)
```

Comment:
@geowurster instead of writing to the repository, let's use the `data` fixture.

```
@pytest.fixture(scope='function')
def data():
    """A temporary directory containing a copy of the files in data."""
    tmpdir = py.test.ensuretemp('tests/data')
    for filename in test_files:
        shutil.copy(filename, str(tmpdir))
    return tmpdir
```